### DISCUSSION TITLE
How to read other nodes, along with the subscribed one?

### DISCUSSION THREAD

**Discussion by prs4** [FreeOpcUa/opcua-asyncio#1400] [Q&A] [answered]
Hello,
I would like to read some other nodes beside of the subscribed one on the moment when the datachange_notification method for the subscribed one is called. Eg. here I would like to gather the other nodes values before self._queue.put_nowait and place them in the array [[node, val, data], [node2, val2, data2], ...] for the self._queue.put_nowait:

`def datachange_notification(self, node: Node, val, data: DataChangeNotif):
        super().datachange_notification(node, val, data)
        self._queue.put_nowait([node, val, data])
`

I wonder if I could use the SubscriptionItemData class fields but I new to the asyncua and not sure if this is the right way. May I have any advice?

**Comment by bauermarkus** [comment]
Did you try to send out a signal and use this signal?

**Reply by prs4** [reply, marked as answer]
You mean read request signal to PLC? No.
BTW I asked the automation system supplier and they said that I don't need to send such a request. Their PLC (S7-1500) works as a OPC UA server and after the subscribed variable change all the other data are available to be read.
Well, I followed an advice to not process to much in 'datachange_notification' method so there I only put new value in a queue. If I can read some other data from the PLC it solves the problem.
